### DISCUSSION TITLE
Extending JWT Builder class to utilize setClaim() not working.

### DISCUSSION THREAD

**Discussion by JMoodyFWD** [phalcon/cphalcon#15321] [Q&A] [answered]
Hey everyone,

I'm trying to extend the JWT Builder class to allow a workaround method of _addClaim()_ to utilize the currently private _setClaim()_ method. I tried utilizing talal424's answer on the old forums of extending the builder class to create a method that will tap into the builder's native method of setting custom claims, but I am still getting a fatal error generated that the _setClaim()_ method is private. Perhaps I'm missing something?

Here is my extension class:
```php
use Phalcon\Security\JWT\Builder;

class CustomJWTBuilder extends Builder
{
    public function addClaim(string $name, $value): Builder
    {
        return $this->setClaim($name, $value);
    }
}
```

And here is how I'm utilizing it:
```php

                $builder = new CustomJWTBuilder($signer);
		// Setup
		$builder
		    ->setAudience('https://targetdomain.tld')    
		    ->setContentType('application/json')          
		    ->setExpirationTime($expires)                 
		    ->setIssuedAt($issued)                        
		    ->setIssuer('https://issuer.tld') 
		    ->setNotBefore($notBefore)                  
		    ->setSubject('Subject')                      
		    ->setPassphrase($passphrase)
		    ->addClaim('Test', '123')
		;

		// Phalcon\Security\JWT\Token\Token object
		$tokenObject = $builder->getToken();

		// The token
		echo $tokenObject->getToken();	
```

This is the error I'm getting:
```
Fatal error: Uncaught Error: Call to private method Phalcon\Security\JWT\Builder::setClaim() from context 'PhalconExt\Security\JWT\CustomJWTBuilder'
```

Any help would be greatly appreciated!

**Comment by niden** [comment, marked as answer]
We will need to change the scope of `setClaim` because the extended class cannot _see_ that method. 

@JMoodyFWD if it is not too much trouble can you open an issue and cc me on it so that I can resolve this?

Thanks!

**Reply by JMoodyFWD** [reply]
Created an issue for this: [https://github.com/phalcon/cphalcon/issues/15322](https://github.com/phalcon/cphalcon/issues/15322)

My apologies if I did it wrong or poorly. First time creating an issue.

**Reply by niden** [reply]
> Created an issue for this: #15322
> 
> My apologies if I did it wrong or poorly. First time creating an issue.

No apology necessary. Thank you for reporting this.

**Reply by JMoodyFWD** [reply]
> > Created an issue for this: #15322
> > 
> > My apologies if I did it wrong or poorly. First time creating an issue.
> 
> No apology necessary. Thank you for reporting this.

My pleasure. Thank you for taking care of this so quickly!

As a slightly off-topic question, I discovered this on my Windows machine where I'm doing my development at the moment. Will there be an updated windows DLL posted soon with this fix included, or is there a fairly simple way that I can compile it for PHP on Windows?

**Reply by Jeckerson** [reply]
![image](https://user-images.githubusercontent.com/3289702/109702955-d12da700-7b8c-11eb-9068-bc38f01c0af3.png)

You can pick an artifact
https://github.com/phalcon/cphalcon/actions/runs/615046427

**Reply by JMoodyFWD** [reply]
> You can pick an artifact
> https://github.com/phalcon/cphalcon/actions/runs/615046427

Awesome! Wasn't aware this was a thing that existed. Thank you again so much for the help! 😄
